src/plugins/discover/public/application/apps/main/components/chart/discover_chart.tsx

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import React, { useCallback } from 'react';
+import moment from 'moment';
+import { EuiFlexGroup, EuiFlexItem, EuiButtonEmpty, EuiSpacer } from '@elastic/eui';
+import { IUiSettingsClient } from 'kibana/public';
+import { i18n } from '@kbn/i18n';
+import { HitsCounter } from '../hits_counter';
+import { DataPublicPluginStart, IndexPattern, search } from '../../../../../../../data/public';
+import { TimechartHeader } from '../timechart_header';
+import { SavedSearch } from '../../../../../saved_searches';
+import { AppState, GetStateReturn } from '../../services/discover_state';
+import { TimechartBucketInterval } from '../timechart_header/timechart_header';
+import { Chart as IChart } from './point_series';
+import { DiscoverHistogram } from './histogram';
+
+const TimechartHeaderMemoized = React.memo(TimechartHeader);
+const DiscoverHistogramMemoized = React.memo(DiscoverHistogram);
+export function DiscoverChart({
+  config,
+  data,
+  bucketInterval,
+  chartData,
+  hits,
+  isLegacy,
+  resetQuery,
+  savedSearch,
+  state,
+  stateContainer,
+  timefield,
+}: {
+  config: IUiSettingsClient;
+  data: DataPublicPluginStart;
+  bucketInterval: TimechartBucketInterval;
+  chartData: IChart;
+  hits: number;
+  indexPattern: IndexPattern;
+  isLegacy: boolean;
+  resetQuery: () => void;
+  savedSearch: SavedSearch;
+  state: AppState;
+  stateContainer: GetStateReturn;
+  timefield?: string;
+}) {
+  const toggleHideChart = useCallback(() => {
+    const newState = { ...state, hideChart: !state.hideChart };
+    stateContainer.setAppState(newState);
+  }, [state, stateContainer]);
```

Comment:
goodnews here: we can just do 

```TypeScript
    stateContainer.setAppState({ hideChart: !state.hideChart });
```
Since we anyway pass in a partial state, it's not necessary to duplicate this merge. thx, better this way